flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -116,6 +117,10 @@
 	 * accessing this don't block the job manager actor and run asynchronously. */
 	private final CompletedCheckpointStore completedCheckpointStore;
 
+	/** The root checkpoint state backend, which is responsible for initializing the
+	 * checkpoint, storing the metadata, and cleaning up the checkpoint */
+	private final StateBackend checkpointStateBackend;
```

Comment:
nit: Are there other backends than "checkpoint" backend?

Reply:
No, not really. Wanted to pick a name here that allows us later to possibly have different "backends" for the data structure, meaning heap and RocksDB in different operators. One state backend needs to be the root that defines how metadata is checkpointed and recovered.